### PR TITLE
Blacklist csv fields

### PR DESCRIPTION
**Pull request by monikasulik**
https://uktrade.atlassian.net/browse/ED-504

### CODE REVIEW

File: supplier/admin.py [hunk 1]
```diff
@@ -47,11 +42,17 @@ def download_csv(self, request, queryset):
             )
         )
 
+        fieldnames = [field for field in Supplier._meta.get_all_field_names()
```
Comment (konradko) [inline supplier/admin.py]: `get_all_field_names` is deprecated in Django 1.10 (https://docs.djangoproject.com/en/1.10/ref/models/meta/), please use the new way if possible

File: supplier/tests/test_admin.py [hunk 2]
```diff
@@ -11,20 +15,21 @@
...
+    'company__logo,company__name,company__number,company__revenue,'
+    'company__sectors,company__website,company_email,'
+    'company_email_confirmed,company_id,date_joined,is_active,'
+    'mobile_number,name,sso_id'
 )
 
 
-class DownloadCSVTestCase:
+@pytest.mark.django_db
+class DownloadCSVTestCase(TestCase):
```
Comment (konradko) [inline supplier/tests/test_admin.py]: woah, good catch, lucky it ever worked

File: supplier/admin.py [hunk 3]
```diff
@@ -13,27 +14,21 @@ class SupplierAdmin(admin.ModelAdmin):
...
-        'company__is_published'
+    csv_excluded_fields = (
+        'id',
+        'company',
+        'created',
+        'modified',
+        'company_email_confirmation_code',
+        'company__supplier_case_studies',
+        'company__suppliers',
+        'company__verified_with_code',
```
Comment (konradko) [inline supplier/admin.py]: I think we want to include this field

File: supplier/admin.py [hunk 4]
```diff
@@ -13,27 +14,21 @@ class SupplierAdmin(admin.ModelAdmin):
...
+        'company__is_verification_letter_sent',
```
Comment (konradko) [inline supplier/admin.py]: ditto

File: supplier/admin.py [hunk 5]
```diff
@@ -13,27 +14,21 @@ class SupplierAdmin(admin.ModelAdmin):
...
+        'company__contact_details',
```
Comment (konradko) [inline supplier/admin.py]: this possibly too

File: supplier/admin.py [hunk 6]
```diff
@@ -13,27 +14,21 @@ class SupplierAdmin(admin.ModelAdmin):
...
+        'company__verification_code',
+        'company__created',
```
Comment (konradko) [inline supplier/admin.py]: ditto

File: supplier/admin.py [hunk 7]
```diff
@@ -13,27 +14,21 @@ class SupplierAdmin(admin.ModelAdmin):
...
+        'company__modified',
```
Comment (konradko) [inline supplier/admin.py]: ditto

**Comment by konradko** [comment]
@monikasulik will the CSV include all the fields that are currently present after this change?

**Comment by monikasulik** [comment]
Yep, it will include all fields from Supplier and Company unless explicitly excluded. Every time a field is added to either of those models the csv tests will fail (which means people will have to give it a thought about whether the field should or shouldn't be in the csv - I think that's good).
The fields are sorted alphabetically, so that there's some sort of consistency to the order in which they appear in the csv. Unfortunately control over field order is one thing we lose with this implementation.

**Comment by konradko** [comment]
@monikasulik good thing with the test failing on model changes. It will be a bit annoying, but at least visible.

File: company/tests/test_models.py [hunk 8]
```diff
@@ -65,7 +65,7 @@ def test_company_case_study_non_required_fields(company):
 
 
 def test_company_model_has_update_create_timestamps():
-    field_names = models.Company._meta.get_all_field_names()
+    field_names = [field.name for field in models.Company._meta.get_fields()]
```
Comment (konradko) [inline company/tests/test_models.py]: 👍

File: supplier/tests/test_admin.py [hunk 9]
```diff
@@ -1,30 +1,43 @@
...
+    'company__number,company__revenue,company__sectors,'
+    'company__twitter_url,company__verified_with_code,'
+    'company__website,company_email,company_email_confirmed,'
+    'date_joined,is_active,mobile_number,name,sso_id'
 )
 
 
-class DownloadCSVTestCase:
+COMPANY_DATA = VALID_REQUEST_DATA.copy()
+# Order in dict is unpredictable so for easier testing just 1 element
```
Comment (konradko) [inline supplier/tests/test_admin.py]: or you can use `OrderedDict` :P

Comment (monikasulik) [inline supplier/tests/test_admin.py]: True. Though as a lot of tests rely on this, can't be sure what the consequences would be. I need factory boy thingies ;)

**Pull request merged**